### PR TITLE
[NOD-669] Rename start/endHash -> low/highHash

### PR DESCRIPTION
**Pull request by stasatdaglabs**

### CODE REVIEW

File: blockdag/dag.go [hunk 1]
```diff
@@ -1472,44 +1472,44 @@ func (dag *BlockDAG) IsInSelectedParentChain(blockHash *daghash.Hash) bool {
...
+// up to the virtual (exclusive). If lowHash is nil then the genesis block is used. If
+// lowHash is not within the select parent chain, go down its own selected parent chain,
 // while collecting each block hash in removedChainHashes, until reaching a block within
 // the main selected parent chain.
 //
 // This method MUST be called with the DAG lock held
-func (dag *BlockDAG) SelectedParentChain(startHash *daghash.Hash) ([]*daghash.Hash, []*daghash.Hash, error) {
-	if startHash == nil {
-		startHash = dag.genesis.hash
+func (dag *BlockDAG) SelectedParentChain(lowHash *daghash.Hash) ([]*daghash.Hash, []*daghash.Hash, error) {
```
Comment (someone235) [inline blockdag/dag.go]: I think you can still use startHash here, or at least change it to blockHash. I don't like the name lowHash without a correspondent highHash.

File: blockdag/dag.go [hunk 2]
```diff
@@ -1610,44 +1610,44 @@ func (dag *BlockDAG) getBlueBlocksHashesBetween(startHash, stopHash *daghash.Has
...
+	highNode := dag.index.LookupNode(highHash)
+	if highNode == nil {
+		return nil, errors.Errorf("Couldn't find high hash %s", highHash)
 	}
 
 	// In order to get no more then maxEntries of blue blocks from
-	// the future of the start node (including itself), we iterate
-	// the selected parent chain of the stopNode and add the blues
-	// each node (including the stopNode itself). This is why the
+	// the future of the low node (including itself), we iterate
```
Comment (someone235) [inline blockdag/dag.go]: change to lowNode

File: blockdag/dag.go [hunk 3]
```diff
@@ -1656,28 +1656,28 @@ func (dag *BlockDAG) getBlueBlocksBetween(startHash, stopHash *daghash.Hash, max
...
 
-// GetBlueBlocksHashesBetween returns the hashes of the blue blocks after the
-// provided start hash until the provided stop hash is reached, or up to the
+// BlueBlocksHashesBetween returns the hashes of the blue blocks after the
+// provided low hash until the provided high hash is reached, or up to the
 // provided max number of block hashes.
 //
 // This function is safe for concurrent access.
-func (dag *BlockDAG) GetBlueBlocksHashesBetween(startHash, stopHash *daghash.Hash, maxHashes uint64) ([]*daghash.Hash, error) {
+func (dag *BlockDAG) BlueBlocksHashesBetween(lowHash, highHash *daghash.Hash, maxHashes uint64) ([]*daghash.Hash, error) {
```
Comment (svarogg) [inline blockdag/dag.go]: I think the gramatically correct name would be `BlueBlockHashesBetween` (without the s in blocks).
The same for `blueBlocksHashesBetween`

File: blockdag/dag.go [hunk 4]
```diff
@@ -1656,28 +1656,28 @@ func (dag *BlockDAG) getBlueBlocksBetween(startHash, stopHash *daghash.Hash, max
...
-// getBlueBlocksHeadersBetween returns the headers of the blue blocks after the
-// provided start hash until the provided stop hash is reached, or up to the
+// blueBlocksHeadersBetween returns the headers of the blue blocks after the
+// provided low hash until the provided high hash is reached, or up to the
 // provided max number of block headers.
 //
 // This function MUST be called with the DAG state lock held (for reads).
-func (dag *BlockDAG) getBlueBlocksHeadersBetween(startHash, stopHash *daghash.Hash, maxHeaders uint64) ([]*wire.BlockHeader, error) {
-	nodes, err := dag.getBlueBlocksBetween(startHash, stopHash, maxHeaders)
+func (dag *BlockDAG) blueBlocksHeadersBetween(lowHash, highHash *daghash.Hash, maxHeaders uint64) ([]*wire.BlockHeader, error) {
```
Comment (svarogg) [inline blockdag/dag.go]: Same here

File: blockdag/dag.go [hunk 5]
```diff
@@ -1734,14 +1734,14 @@ func (dag *BlockDAG) RUnlock() {
...
 
-// GetBlueBlocksHeadersBetween returns the headers of the blocks after the provided
-// start hash until the provided stop hash is reached, or up to the
+// BlueBlocksHeadersBetween returns the headers of the blocks after the provided
+// low hash until the provided high hash is reached, or up to the
 // provided max number of block headers.
 //
 // This function is safe for concurrent access.
-func (dag *BlockDAG) GetBlueBlocksHeadersBetween(startHash, stopHash *daghash.Hash) ([]*wire.BlockHeader, error) {
+func (dag *BlockDAG) BlueBlocksHeadersBetween(lowHash, highHash *daghash.Hash) ([]*wire.BlockHeader, error) {
```
Comment (svarogg) [inline blockdag/dag.go]: And here

**Pull request merged**